tests/integration/test_charm.py

```diff
@@ -88,6 +88,7 @@ async def test_build_and_deploy(ops_test: OpsTest, local_charm: Path) -> None:
     await ops_test.model.wait_for_idle(
         apps=[HYDRA_APP, DB_APP, TRAEFIK_PUBLIC_APP, TRAEFIK_ADMIN_APP],
         raise_on_blocked=False,
+        raise_on_error=False,
         status="active",
         timeout=5 * 60,
     )
```

Comment:
nit: split this into 2 waits, one for external dependencies with `raise_on_error=False` and one only for hydra with `raise_on_error=True`:
```suggestion
    await asyncio.gather(
        await ops_test.model.wait_for_idle(
            apps=[DB_APP, TRAEFIK_PUBLIC_APP, TRAEFIK_ADMIN_APP],
            raise_on_blocked=False,
            raise_on_error=False,
            status="active",
            timeout=5 * 60,
        ),
        await ops_test.model.wait_for_idle(
            apps=[HYDRA_APP],
            raise_on_blocked=False,
            status="active",
            timeout=5 * 60,
        ),
    )
```